homeassistant/components/image/__init__.py

```diff
@@ -295,3 +298,75 @@ async def handle(
             raise web.HTTPInternalServerError() from ex
 
         return web.Response(body=image.content, content_type=image.content_type)
+
+
+async def async_get_still_stream(
+    request: web.Request,
+    image_cb: Callable[[], Awaitable[bytes | None]],
+    content_type: str,
+    interval: float,
+) -> web.StreamResponse:
+    """Generate an HTTP MJPEG stream from the Image.
+
+    This method must be run in the event loop.
+    """
+    response = web.StreamResponse()
+    response.content_type = CONTENT_TYPE_MULTIPART.format("--frameboundary")
+    await response.prepare(request)
+
+    async def write_to_mjpeg_stream(img_bytes: bytes) -> None:
+        """Write image to stream."""
+        await response.write(
+            bytes(
+                "--frameboundary\r\n"
+                f"Content-Type: {content_type}\r\n"
+                f"Content-Length: {len(img_bytes)}\r\n\r\n",
+                "utf-8",
+            )
+            + img_bytes
+            + b"\r\n"
+        )
+
+    last_image = None
+
+    while True:
+        img_bytes = await image_cb()
+        if not img_bytes:
+            break
+
+        if img_bytes != last_image:
+            await write_to_mjpeg_stream(img_bytes)
+
+            # Chrome seems to always ignore first picture,
+            # print it twice.
+            if last_image is None:
+                await write_to_mjpeg_stream(img_bytes)
+            last_image = img_bytes
+
+        await asyncio.sleep(interval)
```

Comment:
As discussed on Discord, I'm not a fan of the fixed interval. We should push updates to the receiver when the image entity is updated.

Reply:
The interval seems to be coming from the client side - I'm still trying to figure it out